ArduCopter/control_guided.cpp

```diff
@@ -212,6 +213,30 @@ void Copter::guided_set_angle(const Quaternion &q, float climb_rate_cms)
     guided_angle_state.update_time_ms = millis();
 }
 
+// guided_reached_destination_notify - notify if we've reached the destination
+// on guided mode - if we're taking off we consider altitude
+void Copter::guided_reached_destination_notify(bool check_altitude)
+{
+    bool reached_alt = true;
+
+    if (guided_prev_mode != guided_mode)
+        guided_reached_notified = false;
```

Comment:
What is the prev_mode for? Seems like there's got to be a better way to detect TO vs WP transitions than that. Also, it needs { } even for the one line.